controllers/dependencies.go

```diff
@@ -31,7 +30,7 @@ func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(meta metav1.O
 		URL: OPERATOR_PIPELINES_REPO,
 	})
 	defer r.removePipelineDependencyFiles(REPO_CLONE_PATH)
-	if err != nil {
+	if err != nil && err != git.ErrRepositoryAlreadyExists {
```

Comment:
Is PlainClone smart enough to check that the existing repo has diverged from the upstream repo? I want to make sure that we always have the most updated repo. If we ignore the AlreadyExist error AND the PlainClone does not check if the git is synced with upstream, there is a chance that we don't pull the latest code. 
If PlainClone does not check whether the git repo is synced, I'd prefer the operator to fail rather than continuing to run.

I am also wondering why we need to handle this. We have a defer method which cleans the git repo not matter what. Did you face any issue with the leftover from the previous operator execution?

Reply:
PlainClone fails is that `dir` is already on disk, sometimes reconcilation happens before the defer is finished. So changed this code to always remove the tmp `dir` which is not error prone since it does not matter if the `dir` exists or not, remove does not throw an error.

Reply:
Ended up making a `tmp` directory to not have to worry about directories anymore